pkg/resource/plugin/analyzer_plugin.go

```diff
@@ -305,3 +313,46 @@ func convertDiagnostics(protoDiagnostics []*pulumirpc.AnalyzeDiagnostic) ([]Anal
 
 	return diagnostics, nil
 }
+
+func constructEnv(opts *PolicyAnalyzerOptions) ([]string, error) {
+	env := os.Environ()
+
+	maybeAppendEnv := func(k, v string) {
+		if v != "" {
+			env = append(env, k+"="+v)
+		}
+	}
+
+	config, err := constructConfig(opts)
+	if err != nil {
+		return nil, err
+	}
+	maybeAppendEnv("PULUMI_CONFIG", config)
```

Comment:
... is this safe?!? if the configuration contains secrets, are they going to be in plain text? And if so, would passing them as environment variables potentially leak them to other processes on the system?

Reply:
Yes, it can contain unencrypted secrets. FWIW, this is what we do to pass config to all our language hosts. For example:

https://github.com/pulumi/pulumi/blob/048acc24f774fa51916d418c45ccc31a05f3f250/sdk/dotnet/cmd/pulumi-language-dotnet/main.go#L512-L548

Reply:
If that is an intentional decision, do we want to document/state that in the comments somehow? So it doesn't otherwise look like an unintentional security problem? (Including config values in plain text is probably OK, for the same reason we pass secret resource outputs. It just might be surprising, that's all.)